pkg/controllers/openshift-apiserver.go

```diff
@@ -120,10 +120,15 @@ func (s *OCPAPIServer) Run(ctx context.Context, ready chan<- struct{}, stopped c
 	}
 
 	stopCh := make(chan struct{})
-	if err := s.options.RunAPIServer(stopCh); err != nil {
-		klog.Fatalf("Failed to start ocp-apiserver %v", err)
-	}
+	defer close(stopCh)
+
+	go func() {
+		if err := s.options.RunAPIServer(stopCh); err != nil {
```

Comment:
Why create and pass in a new stop channel here, shouldn't this be `ctx.Done()`?

Reply:
As much as I looked I didn0t find the openshift-apiserver receiving the context in any form, so we can't cancel it that way. They use a stop channel. Hopefully this would change in a future.

Reply:
That's why I wait for the context, outside, and propagate down to the stop channel.

Reply:
As we talked OOB I'm extracting the stop channel with ctx.Done() for simpler code.